tests/unittests/classification/test_group_fairness.py

```diff
@@ -0,0 +1,127 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from functools import partial
+
+import numpy as np
+import pytest
+import torch
+from fairlearn.metrics import demographic_parity_ratio, true_positive_rate_ratio
+from scipy.special import expit as sigmoid
+
+from torchmetrics.classification.group_fairness import BinaryFairness
+from torchmetrics.functional.classification.group_fairness import binary_fairness
+from unittests.classification.inputs import _group_cases
+from unittests.helpers import seed_all
+from unittests.helpers.testers import THRESHOLD, MetricTester, inject_ignore_index, remove_ignore_index_groups
+
+seed_all(42)
+
+
+def _fairlearn_binary(preds, target, groups, ignore_index):
+    preds = preds.numpy()
+    target = target.numpy()
+    groups = groups.numpy()
+
+    if np.issubdtype(preds.dtype, np.floating):
+        if not ((0 < preds) & (preds < 1)).all():
+            preds = sigmoid(preds)
+        preds = (preds >= THRESHOLD).astype(np.uint8)
+
+    target, preds, groups = remove_ignore_index_groups(target, preds, groups, ignore_index)
+
+    dp = demographic_parity_ratio(target, preds, sensitive_features=groups)
+    eo = true_positive_rate_ratio(target, preds, sensitive_features=groups)
+
+    # TODO: need to add groups here
+    return {"DP": dp, "EO": eo}
```

Comment:
The fairlearn package (and, by extension, most packages that compute group fairness metrics) return a ``float`` with the metric without indicating which groups have a min and max score. If we agree that it would be a good addition, I will write some "custom" code to figure out the min and max scores to add them to the dict keys.